comment/views/flags.py

```diff
@@ -0,0 +1,49 @@
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponseBadRequest, JsonResponse
+from django.shortcuts import get_object_or_404
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext_lazy as _
+from django.views import View
+from django.views.decorators.http import require_POST
+
+from comment.models import Comment, Flag, FlagInstance
+
+
+@method_decorator(require_POST, name='dispatch')
+class SetFlag(LoginRequiredMixin, View):
+
+    @staticmethod
+    def get_flag_object(comment):
```

Comment:
Can you please move this method to Flag manager? I believe this logic is in the wrong place.
Getting an object should be from the model class not the view.

Reply:
I had though of placing this inside the Comment manager or model but that would cause circular imports. Would it be best placing this inside the flag model, design-wise?

Reply:
Sure. The best place for a method that returns a model object is the model manager.
```django
Flag.objects.get_for_comment(comment=comment)
Reaction.objects.get_for_comment(comment=comment)
```

Reply:
I have made the change. In an ideal world, you would want to put the logic inside Comment manager because these objects are related to the instance of the comment model but that creates issues of circular import.

Reply:
Why would the circular import happen? To get the model class inside the manager you can use `self.model` instead of importing it.

Reply:
We need to use the Reaction or Flag class inside the Comment Manager. It would create circular imports as Comment is being used in Reaction and now we can't use Reaction inside Comment.